types/d3/v3/index.d.ts

```diff
@@ -2576,8 +2577,8 @@ declare namespace d3 {
             tickPadding(): number;
             tickPadding(padding: number): Axis;
 
-            tickFormat(): (t: any) => string;
-            tickFormat(format: (t: any) => string): Axis;
+            tickFormat(): (t: any , index : number) => string;
```

Comment:
I feel like I'm missing something obvious, but is `index` optional or not?

Reply:
Hi casey, can you please refer previous comments. You may get the idea of why i did it.

Reply:
I'm assuming you're referring to this comment: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/35994#discussion_r291305510.  I saw it, but didn't follow - could you possibly paraphrase?

Reply:
Don’t use optional parameters in callbacks unless you really mean it:

/* WRONG */
interface Fetcher {
    getObject(done: (data: any, elapsedTime?: number) => void): void;
}

This has a very specific meaning: the done callback might be invoked with 1 argument or might be invoked with 2 arguments. The author probably intended to say that the callback might not care about the elapsedTime parameter, but there’s no need to make the parameter optional to accomplish this – it’s always legal to provide a callback that accepts fewer argument.

What i understood from here is. It's good to keep it as not optional since it provide reasonable value to the function. It's totally up to us to decide whether it should be optional or not. After considering the comment and mapping it with newer type. I decided to change it to not optional. Hope this make sense.